src/riscv-integration.adoc

```diff
@@ -830,13 +837,24 @@ CHERI violations have the following order in priority:
 . Invalid address violation
 . Bounds violation (_Lowest_)
 
-===== mtval2 values for Load Page Faults
+===== mtval2 values for Load/Store/AMO Page Faults
 
-If a load page fault is caused by a CHERI <<cheri_pte_ext,PTE>> fault, then set bit zero of <<mtval2>> to 1 and all other bits to 0, otherwise set all bits to zero.
+Page faults can be caused by normal RISC-V page faults and also by CHERI <<cheri_pte_ext,PTE>> faults.
+If both are detected at once, then both are recorded.
 
-===== mtval2 values for Store Page Faults
+.mtval2 for page faults
+[#mtval2-page-fault,width=70%,float="center",align="center",cols="2,1",options=header]
+|==============================================================================
+| Fault                                                   | Value
+| RISC-V page fault                                       | 0
+| CHERI <<cheri_pte_ext,PTE>> fault                       | 1
+| RISC-V page fault and CHERI <<cheri_pte_ext,PTE>> fault | 2
+|==============================================================================
 
-If a store page fault is caused by a CHERI <<cheri_pte_ext,PTE>> fault, then set bit zero of <<mtval2>> to 1 and all other bits to 0, otherwise set all bits to zero.
+NOTE: Reporting both allows the software the choice about which action to take first, for example a write to a
```

Comment:
This confuses me. Load-side revocation will never do any sweeping on stores, so value 2 isn't useful, it will be treated the same as 1. As for loads, yes, sweeping occurs there, but there's nothing else really to do if both faults occur (for non-tag-dependent implementations) over and above just a CHERI PTE fault, and if your implementation is tag-dependent then the RISC-V page fault and CHERI PTE fault are strictly serial so it's impossible to have both at once.